language/tools/test_generation/src/abstract_state.rs

```diff
@@ -2,36 +2,77 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use std::collections::HashMap;
-use vm::file_format::SignatureToken;
+use vm::file_format::{Kind, SignatureToken};
 
 /// The BorrowState denotes whether a local is `Available` or
 /// has been moved and is `Unavailable`.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq)]
 pub enum BorrowState {
     Available,
     Unavailable,
 }
 
+/// This models a value on the stack or in the locals
+#[derive(Debug, Clone, PartialEq)]
+pub struct AbstractValue {
+    /// Represents the type of the value
+    pub token: SignatureToken,
```

Comment:
It is kind of weird to call a type a token.

Reply:
Unfortunately it's not quite just a type. The token contains in some cases some non-type information. That's why I say it "represents" the type of the value rather than just being the type.